docs/index.adoc

```diff
@@ -592,6 +592,115 @@ public class SecuredServiceEndpoint {
 
 For more details see this guide: https://openliberty.io/guides/microprofile-jwt.html[Securing microservices with JWTs]
 
+== Messaging
+
+Applications can pass messages among themselves with the Java Message Service (JMS) APIs. Liberty can act as a server of JMS destinations (queues and topics,) or it can connect to another server to access those destinations--either another Liberty server or IBM MQ.
+
+=== JMS Server
+Liberty can act as a JMS server whether or not it is serving any applications. 
+
+.server.xml
+[source,xml]
+----
+<featureManager>
+    <feature>wasJmsServer-1.0</feature>
+</featureManager>
+
+<messagingEngine>
+    <queue id="libertyQueue" />
+    <topicSpace id="libertyTopic" />
+</messagingEngine>
+----
+
+If you want applications on other servers to connect to your JMS server, create a `wasJmsEndpoint`. Without this, only applications served by the same Liberty server can acccess the JMS destinations.
+
+.server.xml
+[source,xml]
+----
+<wasJmsEndpoint id="InboundJmsCommsEndpoint" 
+                host="*" 
+                wasJmsPort="7276" 
+                wasJmsSSLPort="7286" />
+----
+
+=== JMS Client
+Liberty can allow applications to connect to both local and remote JMS destinations. (Local connections will also need the JMS server config from the previous section.)
+
+.server.xml
+[source,xml]
+----
+<featureManager>
+    <feature>wasJmsClient-2.0</feature>
+</featureManager>
+
+<jmsConnectionFactory jndiName="jms/localCF">
+    <properties.wasJms />
+</jmsConnectionFactory>
+
+<jmsConnectionFactory jndiName="jms/remoteCF">
+    <properties.wasJms remoteServerAddress="example.com:7276:BootstrapBasicMessaging" />
```

Comment:
What does `BootstrapBasicMessaging` in the URL mean here?

Reply:
I'll add a sentence describing the remoteServerAddress format, but basically it's kind of a holdover from tWAS that specifies a non-SSL connection. BootstrapSecureMessaging would specify an SSL connection.